src/memory/kotlin_binding_manager.cpp

```diff
@@ -0,0 +1,112 @@
+#include "kotlin_binding_manager.h"
+#include "modules/kotlin_jvm/src/logging.h"
+
+GDNativeInstanceBindingCallbacks KotlinBindingManager::_instance_binding_callbacks = {
+	&_instance_binding_create_callback,
+	&_instance_binding_free_callback,
+	&_instance_binding_reference_callback
+};
+
+void *KotlinBindingManager::_instance_binding_create_callback(void *p_token, void *p_instance) {
+	KotlinBindingManager &manager = get_instance();
+	Object *owner = reinterpret_cast<Object *>(p_instance);
+
+	manager.spin.lock();
+	KotlinBinding *binding = &manager.binding_map.insert(owner, KotlinBinding())->value;
```

Comment:
Why getting it as a ptr instead of a ref ?

Reply:
The function signature requires to return a pointer.